src/turnkeyml/common/system_info.py

```diff
@@ -0,0 +1,399 @@
+                )
+                .decode()
+                .strip()
+                .replace("\n", " ")
+            )
+            return oem_info
+        except subprocess.CalledProcessError:
+            # This catches the case where sudo requires a password
+            return "Unable to get oem info - password required"
+        except Exception as e:  # pylint: disable=broad-except
+            return f"ERROR - {e}"
+
+    @staticmethod
+    def get_physical_memory() -> str:
+        """
+        Retrieves the physical memory of the computer system.
+
+        Returns:
+            str: Physical memory
+        """
+        try:
+            mem_info = (
+                subprocess.check_output("free -m", shell=True)
+                .decode()
+                .split("\n")[1]
+                .split()[1]
+            )
+            mem_info_gb = round(int(mem_info) / 1024, 2)
+            return f"{mem_info_gb} GB"
+        except Exception as e:  # pylint: disable=broad-except
+            return f"ERROR - {e}"
+
+    def get_dict(self) -> dict:
+        """
+        Retrieves all the system information into a dictionary
+
+        Returns:
+            dict: System information
+        """
+        info_dict = super().get_dict()
+        info_dict["Processor"] = self.get_processor_name()
+        info_dict["OEM System"] = self.get_system_model()
+        info_dict["Physical Memory"] = self.get_physical_memory()
+        return info_dict
+
+
+class UnsupportedOSSystemInfo(SystemInfo):
+    """Class used to access system information in unsupported operating systems"""
+
+    def get_dict(self):
+        """
+        Retrieves all the system information into a dictionary
+
+        Returns:
+            dict: System information
+        """
+        info_dict = super().get_dict()
+        info_dict["Error"] = "UNSUPPORTED OS"
+        return info_dict
+
+
+def get_system_info() -> SystemInfo:
+    """
+    Creates the appropriate SystemInfo object based on the operating system.
+
+    Returns:
+        A subclass of SystemInfo for the current operating system.
+    """
+    os_type = platform.system()
+    if os_type == "Windows":
+        return WindowsSystemInfo()
+    elif os_type == "Linux":
+        # WSL has to be handled differently compared to native Linux
+        if "microsoft" in str(platform.release()):
+            return WSLSystemInfo()
+        else:
+            return LinuxSystemInfo()
+    else:
+        return UnsupportedOSSystemInfo()
+
+
+def get_system_info_dict() -> dict:
+    """
+    Puts the system information into a dictionary.
+
+    Returns:
+        dict: Dictionary containing the system information.
+    """
+    return get_system_info().get_dict()
+
+
+def main():
+    """
+    Prints the system information dictionary.
+    """
+    info_dict = get_system_info_dict()
+    for k, v in info_dict.items():
+        print(f"{k}: {v}")
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
I was able to test this on WSL and it works as expected. But I don't think we need this file to have an entry point.
Can we directly use `get_system_info()` in other modules?

Reply:
Yes.  The entry point was for test purposes and could also be handy to provide a simple way to understand the current environment from the command line.

In `sequence.py`, we simply use:
```
from turnkeyml.common.system_info import get_system_info_dict
.
.
.
        # Save the system information used for this build
        system_info_dict = get_system_info_dict()
        state.save_stat(
            fs.Keys.SYSTEM_INFO,
            system_info_dict,
        )
```

If you don't need all the information and don't want to take the ~5 secs hit to get it all, you can access selective elements via:
```
from turnkeyml.common.system_info import get_system_info
.
.
.
    system_info = get_system_info()
    os_version = system_info.get_os_version()
    python_packages = system_info.get_python_packages()
```

Reply:
Based on our chat, I removed this entry point and added `system-info` as a management tool.  You will now get system information printed out with the command `lemonade system-info`.